### ISSUE TITLE
Application invisible to system and apps

### ISSUE THREAD

**Issue by WeaponsLeft** [microg/android_packages_apps_GmsCore#535]
Hello I face problem, that this application is invisble to nearly all applications on my device.
Due to this I can not unblock it in my firewall as it is not listed.

The follwing application can not detect gms-core in any way:
- Afwall+ (root) can not detect the application
- System buildin "apps manager" can not detect the application
- "system app safe remover" (root) can not detect the application
- "lucky patcher" (root) can not see the application

The only application that can see the package is:
- X-Posed based application "quick settings"

I already tried the follwing solutions for testing purposes:
- Install application as user application
- Install application as privledged application
- Signature spoofing enabled/disabled

Additional information:
MicroG package: Services Core com.google.android.gms -> 0.2.4-111-gf1cdb48
Original file: https://microg.org/fdroid/repo/com.google.android.gms-12221403.apk
Device: "Zuk Z2 Pro" / Model: "ZUK Z2121"
Device operating system "OREO (8.0.0)"
OS release: ZUI 3.5.344 ST

Any ideas why this application goes under the radar?
Is this normal?

**Comment by WeaponsLeft** [comment]
Is this a fun question like holding a ball in the hands an getting asked how to determine it is not a cube? 😁

Because all of those applications scan installed packages and present them as a list so one ore more packages can be selected to perfom an action on them. However, none of those applications (except the one I mentioned) includes "com.google.android.gms", "org.microg.gms", "microG Settings" or "microG Services Core" in any of the lists. As I mentioned, even the buildin apps manager of the operating system does not list the application as installed, thus the application is literarely invisible. Due to this I can not unblock it in Afwall+ firewall as it never shows up in my white-list which makes using several functions of microG impossible because it can not access the internet.

How to solve it?

**Comment by ArchangeGabriel** [comment]
This is kinda what I expected. Does any of them present “Google Play Services”?

**Comment by WeaponsLeft** [comment]
Unfortunately no "Google Play Services" is shown either. You mean because it could be possible shown as "Google Play Services" due to the signature spoofing?
Already tried with signature spoofing disabled but that makes no difference.

**Comment by ArchangeGabriel** [comment]
Yes, exactly. That’s how it shows in Yalp Store for me for instance. Have you looked at “UnifiedNlp Location Services” or anything with μG/microG in the name?

Honestly excepted apparition under another name, I don’t see what could be happening.

**Comment by WeaponsLeft** [comment]
**I have the following applications installed:**
- Services Core (v0.2.4-111-gf1-cdb48)
- Services Framework Proxy (v0.1.0)
- Store - FakeStore release (v0.0.2) 
- DroidGuard Helper (v0.1.0-10-gf64bf69)
- IchnaeaNlpBackend (v1.3.3)
- NominatimGeocoderBackend (v1.2.1)
- FakeGapps (v2.0) [xposed-module for signature spoofing]
- XposedGmsCoreUnifiedNlp (v1.0.4-Beta) [xposed-module to allow unifiednlp as location provider]
- Yalp Store (v0.39)

**MicroG self-check:**
![microg-selfcheck](https://user-images.githubusercontent.com/10565802/39260490-fb7bf088-48b9-11e8-96ed-6381c09d259b.png)

**Yalp Store:**
![yalp-application-list](https://user-images.githubusercontent.com/10565802/39261039-849c33cc-48bb-11e8-8748-3f7ef5c1fc9d.png)

**Comment by ArchangeGabriel** [comment]
Well I don’t know. I might be because of how the XPosed module works, it could be something with Magisk… This is not something I can determine as just an user.

**Comment by wombatch** [comment]
I am no expert here, as can be seen from my other posts, but there is a xposed module (imaginatively AFWall+ Xposed related fixes for AFWall+) to allow AFwall to work. I have it installed and my AFWall+ (v2.9.8)  can see "microG Services Core".
